custom-configs/src/main/java/com/sequenceiq/cloudbreak/domain/CustomConfigs.java

```diff
@@ -0,0 +1,141 @@
+package com.sequenceiq.cloudbreak.domain;
+
+import java.io.Serializable;
+import java.util.Set;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@Entity
+@Table(
+        uniqueConstraints = @UniqueConstraint(columnNames = {"name", "resourcecrn"})
+)
+public class CustomConfigs implements Serializable {
+    @Id
+    @SequenceGenerator(
+            name = "custom_configs_generator",
+            sequenceName = "custom_configs_id_seq",
+            allocationSize = 1
+    )
+    @GeneratedValue(
+            strategy = GenerationType.SEQUENCE,
+            generator = "custom_configs_generator"
+    )
+    @Column
+    private Long id;
+
+    @Column
+    private String name;
+
+    @Column
+    private String resourceCrn;
+
+    @OneToMany(mappedBy = "customConfigs", cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
+    private Set<CustomConfigProperty> configs;
+
+    @Column
+    private String platformVersion;
+
+    @Column
+    private String account;
+
+    @Column
+    private Long created = System.currentTimeMillis();
+
+    public CustomConfigs(String name, String resourceCrn, Set<CustomConfigProperty> configs, String platformVersion, String account,
+            Long created) {
+        this.name = name;
+        this.resourceCrn = resourceCrn;
+        this.configs = configs;
+        this.platformVersion = platformVersion;
+        this.account = account;
+        this.created = created;
+    }
+
+    public CustomConfigs(CustomConfigs existingCustomConfigs) {
+        this.name = existingCustomConfigs.getName();
+        this.configs = existingCustomConfigs.getConfigs();
+        this.platformVersion = existingCustomConfigs.getPlatformVersion();
+    }
```

Comment:
Since this is a copy constructor you should copy the configs also and not reusing existing set.